templates/serverpod_templates/github/workflows/deployment-docker.yml

```diff
@@ -0,0 +1,125 @@
+# This is a workflow to deploy your Serverpod to a single machine using docker-compose. This is useful for testing and small deployments. For larger deployments, you should use the deployment-aws.yml or deployment-gcp.yml workflows.
+# To reduce the workload on the machine we do not use redis in this deployment. If you want to use redis, you need to add it to the docker-compose file and the serverpod configuration.
+# You need to setup the correct hostnames in the docker-compose-production file AND the serverpod configuration file.
+# Before deploying this workflow, you need to set up the following secrets in your repository settings:
+# PAT_GITHUB: A GitHub token with the repo scope
+# PAT_USER_GITHUB: GitHub username for the PAT
+# SERVERPOD_PASSWORDS: The passwords file for the serverpod (simple copy & paste)
+#
+# ## Server setup ##
+# on your vps / server you need to generate a keypair and add the public key to the authorized_keys file
+## ssh-keygen -t rsa -b 4096
+## cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+# now copy the private key and add it to the secrets in the github repository
+## cat ~/.ssh/id_rsa
+# SSH_PRIVATE_KEY: The private key for the SSH connection which you just 'cat'ed
+# SSH_HOST: The host for the SSH connection -> IP or domain
+# SSH_USER: The user for the SSH connection -> often root
+# DATABASE_PASSWORD_PRODUCTION: The password for the production database
+
+name: Deploy to docker
+on:
+  push:
+    branches: [deployment-docker-production]
+  workflow_dispatch:
+    inputs:
+      target:
+        description: "Target"
+        required: true
+        default: "production"
+        type: choice
+        options:
+          - "production"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  # TODO Update with your GitHub username or organization
+  GHCR_ORG: "<ORGANIZATION>"
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    defaults:
+      run:
+        working-directory: ./projectname_server
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Create passwords file
+        working-directory: projectname_server
+        shell: bash
+        env:
+          SERVERPOD_PASSWORDS: ${{ secrets.SERVERPOD_PASSWORDS }}
+        run: |
+          pwd
+          echo "$SERVERPOD_PASSWORDS" > config/passwords.yaml
+          ls config/
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
```

Comment:
maybe using the actor is a bad idea?

Reply:
We could use github.repository_owner https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Reply:
But actor is from the official example: https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions